openpype/plugins/publish/collect_context_label.py

```diff
@@ -16,16 +17,22 @@ class CollectContextLabel(pyblish.api.ContextPlugin):
     label = "Context Label"
 
     def process(self, context):
-
-        # Get last registered host
-        host = pyblish.api.registered_hosts()[-1]
-
-        # Get scene name from "currentFile"
-        path = context.data.get("currentFile") or "<Unsaved>"
-        base = os.path.basename(path)
+        # Add ability to use custom context label
+        context_label = context.data.get("contextLabel")
+        if context_label:
+            context.data["label"] = context_label
+            return
+
+        host_name = context.data.get("hostName")
+        if not host_name:
+            host_name = pyblish.api.registered_hosts()[-1]
```

Comment:
Would this still work with the new class based hosts implementations?

Reply:
This is a backup solution when `hostName` is not collected. I think this will never be used in UI.

Reply:
To answer question. The `"hostName"` on context data has some logic how it can be filled and one possible approach can be to use registered host.